pynumdiff/optimize/_optimize.py

```diff
@@ -103,33 +108,43 @@
 
 # This function has to be at the top level for multiprocessing but is only used by optimize.
 def _objective_function(point, func, x, dt, singleton_params, categorical_params, search_space_types,
-    dxdt_truth, metric, tvgamma, padding):
+    dxdt_truth, metric, tvgamma, padding, cache):
     """Function minimized by scipy.optimize.minimize, needs to have the form: (point, *args) -> float
     This is mildly complicated, because "point" controls the settings of a differentiation function, but
     the method may have numerical and non-numerical parameters, and all such parameters are now passed by
     keyword arguments. So the encoded `point` has to be decoded to dict.
 
     :param np.array point: a numerical vector scipy chooses to try in the objective function
-    All other parameters documented in `optimize`
+    :param dict singleton_params: maps parameter names to singleton values
+    :param dict categorical_params: maps parameter names to values
+    :param dict search_space_types: maps parameter names to types, for turning float vector point into dict point_params
+    :param multiprocessing.manager.dict cache: available across processes to save results and work
+    Other params documented in `optimize`
 
     :return: float, cost of this objective at the point
     """
-    point_params = {k:(v if search_space_types[k] == float else 
-                int(np.round(v))) for k,v in zip(search_space_types, point)} # point -> dict
+    key = sha1((''.join(f"{v:.3e}" for v in point) + # This hash is stable across processes. Takes bytes
```

Comment:
Hashing unhashable types reliably was more pain than I expected. It's still not totally perfect, because scientific notation isn't a totally reliable way to see we've already queried somewhere very nearby. For instance, 0.00e+0 looks very different from 1.00e-5 as a string, but 1.10e3 vs identical-looking 1.10e3 could be hiding a difference much greater than 1e-5.